File: app/src/main/java/net/vrallev/kapt3/bug/SomeAutoValueObject.java
package net.vrallev.kapt3.bug;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class SomeAutoValueObject {

  public abstract int getUserId();
  public abstract String getPhotoUrl();
  public abstract String getContactName();
  public abstract long getLastSendDate();

  public static Builder builder() {
    return new AutoValue_SomeAutoValueObject.Builder();
  }

  @AutoValue.Builder
  public abstract static class Builder {
    public abstract Builder setUserId(int newUserId);

    public abstract Builder setPhotoUrl(String newPhotoUrl);

    public abstract Builder setContactName(String newContactName);

    public abstract Builder setLastSendDate(long newLastSendDate);

    public abstract SomeAutoValueObject build();
  }
}

File: app/src/main/kotlin/net/vrallev/kapt3/bug/SomeAutoValueObject.kt
package net.vrallev.kapt3.bug

import com.google.auto.value.AutoValue

@AutoValue
abstract class SomeAutoValueObject {

    abstract val userId: Int
    abstract val photoUrl: String
    abstract val contactName: String
    abstract val lastSendDate: Long

    @AutoValue.Builder
    abstract class Builder {
        abstract fun setUserId(newUserId: Int): Builder

        abstract fun setPhotoUrl(newPhotoUrl: String): Builder

        abstract fun setContactName(newContactName: String): Builder

        abstract fun setLastSendDate(newLastSendDate: Long): Builder

        abstract fun build(): SomeAutoValueObject
    }

    companion object {

        fun builder(): Builder {
            return AutoValue_SomeAutoValueObject.Builder()
        }
    }
}
